plugins/org.locationtech.udig.project/src/org/locationtech/udig/project/internal/impl/LayerImpl.java

```diff
@@ -2484,4 +2484,20 @@ public static int doComparison(ILayer layer, ILayer layer2) {
         return allRanges;
     }
 
+    /**
+     * Returns layer ID taking properly care to hide user password by displaying * chars instead
+     * 
+     * @param layer
+     * @return
+     */
+        public static String getDisplayID(final Layer layer) {
```

Comment:
I'm wondering why the method signiture has not been added to Interface org.locationtech.udig.project.internal.Layer 
Since the interface is intenal, nobody else should have implemented another layer ..  but could be because package org.locationtech.udig.project.internal is exported .. hmm

Any opinions?

Reply:
Probably it could be also have been implemented the way you mention. 
I think I choose to do it this way (static method) due to the fact that Layer inherits from EObject and I did not want to give the impression that this is a member returning function.
If you think it is more clear the way you propose we can change it.

Reply:
@nprigour I had a look at the changset today. I'd like to understand two methods in LayerImpl and IGeoResource which looks like a bit duplicated code to me. Is there any reason why LayerSummary cannot add `layer.getGeoResource().getDisplayID()`? 

The second note I have : Have you used the[ codeformatter.xml](extras/org.locationtech.udig.dev/codeformatter.xml) configuration to format new code? If not, please have a look at https://github.com/locationtech/udig-platform#preferences and format new code blocks you contributed. It helps us to read the code regarding tabs and spaces ;) 

Many thanks